lib/pc_matrix.c

```diff
@@ -61,3 +84,18 @@ pc_matrix_multiply_vector(PCVEC3 res, const PCMAT33 mat, const PCVEC3 vec)
     res[1] = mat[3] * vec[0] + mat[4] * vec[1] + mat[5] * vec[2];
     res[2] = mat[6] * vec[0] + mat[7] * vec[1] + mat[8] * vec[2];
 }
+
+/**
+* Multiply mat by vec and store the resulting vector in res.
+* 4x3 version
+*/
+void
+pc_matrix_multiply_vector_43(PCVEC3 res, const PCMAT43 mat, const PCVEC4 vec)
+{
+    res[0] = mat[0] * vec[0] + mat[1] * vec[1]
+        + mat[2] * vec[2] + mat[3] * vec[3];
+    res[1] = mat[4] * vec[0] + mat[5] * vec[1]
+        + mat[6] * vec[2] + mat[7] * vec[3];
+    res[2] = mat[8] * vec[0] + mat[9] * vec[1]
+        + mat[10] * vec[2] + mat[11] * vec[3];
+}
```

Comment:
vec could be a PCVEC3, indexing issues for mat, multiplication by vec[3]==1 should not be performed